frontend/src/utils/action.tsx

```diff
@@ -23,16 +25,35 @@ export const CompareNowAction = ({ videoId }: { videoId: string }) => {
 
 export const AddToRateLaterList = ({ videoId }: { videoId: string }) => {
   const video_id = videoId;
+  const { enqueueSnackbar } = useSnackbar();
   return (
     <Tooltip title="Rate later" placement="left">
       <IconButton
         size="medium"
         color="secondary"
-        onClick={() =>
-          UsersService.usersMeVideoRateLaterCreate({
-            video: { video_id } as Video,
-          })
-        }
+        onClick={async () => {
+          let flag = false;
+          const rateLaterList = await UsersService.usersMeVideoRateLaterList();
+          rateLaterList.results?.forEach((rateLaterVideo) => {
+            if (rateLaterVideo.video.video_id === video_id) {
+              flag = true;
+            }
+          });
+          if (!flag) {
+            await UsersService.usersMeVideoRateLaterCreate({
+              video: { video_id } as Video,
+            });
+            showSuccessAlert(
+              enqueueSnackbar,
+              'The video has been added to your rate later list.'
+            );
+          } else {
+            showErrorAlert(
+              enqueueSnackbar,
+              'The video is already in your rate later list.'
+            );
```

Comment:
Is that really an error? The message in red might be unnecessarily frightening. Maybe an "info" alert would be more appropriate.